plugins/jobs/src/js/components/JobsOverviewTable.js

```diff
@@ -69,37 +68,68 @@ class JobsTable extends React.Component {
       }
     ];
   }
-
-  // TODO: DCOS-7766 Revisit this pre-rendering data transformation...
-  getData() {
-    return this.props.jobs.map(function(job) {
-      const isGroup = job instanceof Tree;
-      let lastRun = {};
-      let schedules = null;
-      let status = null;
-
-      if (!isGroup) {
-        const lastRunsSummary = job.getLastRunsSummary();
-
-        lastRun = {
-          status: job.getLastRunStatus().status,
-          lastSuccessAt: lastRunsSummary.lastSuccessAt,
-          lastFailureAt: lastRunsSummary.lastFailureAt
+  /**
+   * converts data from Job to table format.
+   *
+   * @param  {JobConnection} data
+   * @returns {Array<{ id, idGroup, name, schedules, status, lastRun }>} object with data needed by Table Component
+   */
+  getData(data) {
+    return Object.values(
+      data.nodes.reduce((acc, job) => {
+        /*
+        * we can find out if current job is nested in another path by
+        * comparing the job.path array with our given data.path
+        *
+        * we know already, that all jobs are in our given data.path -
+        * no need to check for something like bar.bar.baz when data.path is "foo.bar"
+        *
+        * Example:
+        * job.path = "foo.bar.baz";
+        * data.path = "foo.bar";
+        * => isGroup should be true, because there is "baz" as extra path below our prefix
+        */
+        const isGroup = job.path.slice(data.path.length).length > 0;
+        let lastRun = {};
+        let schedules = null;
+        let status = null;
+        let name = "";
```

Comment:
🎨 I would really like us to get rid of the `let` statements. I think it would be far more readable if there were more functions, either accessors that know how to get X (e.g. id or name) no matter if it's a job or a group or a function that gets the information needed for a group plus one that does the same for a job.